### DISCUSSION TITLE
3D object detection and labeling

### DISCUSSION THREAD

**Discussion by PercyMcfly** [CariusLars/ar_flutter_plugin#57] [Q&A]
Hey,

i saw that it‘s pretty easy to detect certain 3D objects with the camera on AR-Kit.

I would like to detect something i have scanned and added to the database, and show a label on top of it. In addition, i would like to be able to tap that label and execute a function.

Is there any way to do that using the ar_flutter_plugin?

**Comment by CariusLars** [comment]
Hey @PercyMcfly ,
you're right, ARKit can do this pretty well, however, the functionality does not exist in ARCore (as far as I'm aware of), so I did not include it in the AR Flutter plugin. Unfortunately, this makes your request not realisable with the current state of the plugin. Is it a requirement for you to go cross platform?
Cheers
Lars

**Reply by PercyMcfly** [reply]
Hey @CariusLars ,

Thats what I suspected. My app is cross platform, but I think for the beginning, it would be okay for this feature to only be available on iOS(12 or higher). 
Do you know if the ARKit plugin allows for this feature? After a quick skim, I wasn't able to find any reference on this either. 

Are you considering building in platform dependent functionality into the ar_flutter_plugin?

Best,
Percy

**Reply by CariusLars** [reply]
Hey @PercyMcfly ,
no sure if the ARKit plugin supports that, it's probably best you ask it in their discussion forum.
Tbh, I'm not really considering building in platform dependent functionality as the core value of this plugin is seamless integration accross both iOS and Android, so the only option would be manually adding this functionality to Android as well which seems highly complicated int his case...
Cheers,
Lars
